cpp/src/barretenberg/honk/proof_system/work_queue.hpp

```diff
@@ -0,0 +1,143 @@
+#pragma once
+
+#include "barretenberg/honk/transcript/transcript.hpp"
+#include "barretenberg/plonk/proof_system/proving_key/proving_key.hpp"
+#include <cstddef>
+#include <memory>
+
+namespace proof_system::honk {
+
+// Currently only one type of work queue operation but there will likely be others related to Sumcheck
+enum WorkType { SCALAR_MULTIPLICATION };
+
+template <typename Params> class work_queue {
+
+    using CommitmentKey = typename Params::CK;
+    using FF = typename Params::Fr;
+    using Commitment = typename Params::C;
+
+    struct work_item_info {
+        uint32_t num_scalar_multiplications;
+    };
+
+    struct work_item {
+        WorkType work_type = SCALAR_MULTIPLICATION;
+        std::span<FF> mul_scalars;
+        std::string label;
+    };
+
+  private:
+    std::shared_ptr<proof_system::plonk::proving_key> key;
```

Comment:
Can we call this `proving_key` since there's another type of key floating around in this class?

Reply:
good call, done

Reply:
Thanks.